### PR TITLE
Secondary voltage control notifications

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

No

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

When a pilot point target voltage or a control unit status is modified, network listeners are not notified.

**What is the new behavior (if this is a feature change)?**
We added an extension notification feature and secondary voltage control has been fully redesigned to support notifications.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [ ] No

**If yes, please check if the following requirements are fulfilled**

- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration steps are described in the following section

**What changes might users need to make in their application due to this PR? (migration steps)**

**Other information**:

### CODE REVIEW

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java [hunk 1]
```diff
@@ -6,33 +6,40 @@
...
 
     void afterRemoval(String id);
 
-    void onUpdate(Identifiable identifiable, String attribute, Object oldValue, Object newValue);
+    void onUpdate(Identifiable<?> identifiable, String attribute, Object oldValue, Object newValue);
 
-    default void onUpdate(Identifiable identifiable, String attribute, String variantId, Object oldValue,
-            Object newValue) {
-        // empty default implementation
-    }
```
Comment (olperr1) [inline iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java:25]: Why did you remove this default implementation but keep the ones for `onElementAdded`, `onElementReplaced` and `onElementRemoved`?

Comment (geofjamg) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/NetworkListener.java:25]: We could also. I will open a PR to also rename them as element is not a right term (this is in really only properties add or remove)

**Review by olperr1 — APPROVED** [review]
